LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/NetworkServerConfiguration.cs

```diff
@@ -138,7 +138,7 @@ public static NetworkServerConfiguration CreateFromEnvironmentVariables()
             config.EnableGateway = envVars.GetEnvVar("ENABLE_GATEWAY", config.EnableGateway);
             config.GatewayID = envVars.GetEnvVar("IOTEDGE_DEVICEID", string.Empty);
             config.HttpsProxy = envVars.GetEnvVar("HTTPS_PROXY", string.Empty);
-            config.Rx2DataRate = envVars.GetEnvVar("RX2_DATR", string.Empty);
+            config.Rx2DataRate = envVars.GetEnvVar("RX2_DATR") is { } someDr ? (DataRateIndex)someDr : null;
```

Comment:
Nice, but then `GetEnvVar` returns a `double` here. Don't think we want that? How about the following:

```suggestion
            config.Rx2DataRate = envVars.GetEnvVar("RX2_DATR", -1) is var datrNum && (DataRateIndex)datrNum is var datr && Enum.IsDefined(datr) ? datr : null;
```

What's our policy on misconfiguration? I think we should throw if, for example, the data rate is in correct (outside of the 0..15 range). If so, should we open a bug for that and potentially other possible cases of misconfiguration?

Reply:
this is an optional property, therefore I think setting it to fail when not set I think is not a good idea

Reply:
What I meant is validating that it's correct _when it is set_.

Reply:
To be elaborated in #1137